pandas/tests/io/formats/test_to_csv.py

```diff
@@ -223,3 +224,46 @@ def test_to_csv_multi_index(self):
 
         exp = "foo\nbar\n1\n"
         assert df.to_csv(index=False) == exp
+
+    def test_to_csv_compression(self):
```

Comment:
follow the style of the read_csv tests for this, IOW you need to parameterize these, skipping as needed